pandas/core/groupby/groupby.py

```diff
@@ -2271,28 +2271,18 @@ def pre_processor(vals: np.ndarray) -> Tuple[np.ndarray, Optional[Type]]:
                     "'quantile' cannot be performed against 'object' dtypes!"
                 )
 
-            inference = None
+            inference: Optional[np.dtype] = None
```

Comment:
the return type is `Tuple[np.ndarray, Optional[Type]]` in the function signature. so that can also now be narrowed.

Reply:
was needed anyway for mypy fixup